cmd/extensions/main.go

```diff
@@ -170,6 +171,16 @@ func main() {
 		health = healthcheck.NewHandler()
 	}
 
+	podReady = true
+	health.AddReadinessCheck("agones-extensions", func() error {
+		if !podReady {
+			return errors.New("Pod is not in Ready state")
+		}
+		return nil
+	})
+
+	signals.NewSigTermHandler(ctx, HandleSigTerm)
```

Comment:
We talked about it previously, but you're going to need to deal with line 97 (the `NewSigKillContext`). What I would probably do is change line 97 to:

```
ctx, cancelCtx := context.WithCancel(context.Background())
```

then in `HandleSigTerm`, you'll need to call that cancelFunc().

Frankly, to make this easier to understand and avoid using global variables, I would consider just inlining the function here, i.e.:

```
signals.NewSigTermHandler(crd, func() {
  podReady = false
  time.Sleep(mumble mumble)
  cancelCtx()
  ???exit???
})
```

but you're on the right track.

Reply:
Ooooh, I see.